### PR TITLE
JSON-RPC Proxy improvements

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: scripts/jsonrpcproxy.py [hunk 1]
```diff
@@ -111,6 +129,9 @@ def __init__(self, ipc_path):
     def is_connected(self):
         return True
 
+    def check_connection(self, timeout):
+        pass
```
Comment (axic) [inline scripts/jsonrpcproxy.py]: Perhaps add a todo comment here?

File: scripts/jsonrpcproxy.py [hunk 2]
```diff
@@ -82,6 +84,22 @@ def close(self):
     def is_connected(self):
         return self._socket is not None
 
+    def check_connection(self, timeout):
+        SLEEPTIME = 0.1
```
Comment (axic) [inline scripts/jsonrpcproxy.py]: Is this in seconds?

File: scripts/jsonrpcproxy.py (hunk 1 above)
Comment (chfast) [reply scripts/jsonrpcproxy.py]: I'm not sure it should be on any todo list. Looks like for named pipes on Windows it is always "connected".

File: scripts/jsonrpcproxy.py (hunk 2 above)
Comment (chfast) [reply scripts/jsonrpcproxy.py]: Yes.

File: scripts/jsonrpcproxy.py [hunk 3]
```diff
@@ -221,6 +238,14 @@ def process(self, request):
 
         return response
 
+    def run(self):
+        self.conn = get_ipc_connector(self.backend_address)
+        self.conn.check_connection(timeout=10.0)
```
Comment (axic) [inline scripts/jsonrpcproxy.py]: There is a `BUFSIZE=32` in the top. Maybe move this as a configuration value up there too?

Comment (axic) [reply scripts/jsonrpcproxy.py]: Also suggest to bump the `VERSION` there.

Comment (chfast) [reply scripts/jsonrpcproxy.py]: Bumped to 0.2

**Pull request merged**